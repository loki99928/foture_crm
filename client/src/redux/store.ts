import {Action, applyMiddleware, combineReducers, compose, createStore} from "redux"
import thankMiddleware, { ThunkAction } from 'redux-thunk'
import AuthReducer from "./Reducer/Auth";
import AppReducer from "./Reducer/App";

let rootReducer = combineReducers({
     AuthReducer,
     AppReducer,
})
export type AppStateType = ReturnType<typeof rootReducer>

// @ts-ignore
const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose

const store = createStore(rootReducer,  composeEnhancers( applyMiddleware(thankMiddleware)))

export type InferActionTypes<T> = T extends {[key: string]: (...args: any[])=> infer U} ? U : never

export type BaseThunkType<A extends Action, R = Promise<any>> = ThunkAction<R, AppStateType, unknown, A>

export default store