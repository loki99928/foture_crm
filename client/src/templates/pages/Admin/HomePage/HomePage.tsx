import React from "react";
import Sidebar from "../../../components/sidebar/default";
import Header from "../../../components/header/default";

const HomePage = () => {
    return (
        <React.Fragment>
            <div className="wrapper container">
                <Header/>
                <div className="row">
                    <Sidebar/>
                    <main className="content">
                        Home
                    </main>
                </div>
            </div>
        </React.Fragment>
    )
}

export default HomePage;