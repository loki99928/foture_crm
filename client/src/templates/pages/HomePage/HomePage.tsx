import React from "react";

import Home from "../../components/content/Home/Home";

export const HomePage = () => {

    return (
        <main>
            <Home/>
        </main>
    )
}