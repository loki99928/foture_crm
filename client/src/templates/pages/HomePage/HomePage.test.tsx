import React from "react";
import {createMemoryHistory} from "history";
import {render} from "@testing-library/react";
import {Router} from "react-router";
import {Provider} from "react-redux";

import store from "../../../redux/store";
import {HomePage} from "./HomePage";

describe('Auth page', () => {
    it('render', function () {
        const hashHistory = createMemoryHistory()

        const { asFragment } = render(
            <Provider store={store}>
                <Router location={'/'} navigator={hashHistory}>
                    <HomePage/>
                </Router>
            </Provider>
        )

        expect(asFragment()).toMatchSnapshot()
    })
})