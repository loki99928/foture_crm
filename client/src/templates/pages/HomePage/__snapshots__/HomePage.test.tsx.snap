// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auth page render 1`] = `
<DocumentFragment>
  <main />
</DocumentFragment>
`;
